docs/filesystem.rst

```diff
@@ -0,0 +1,179 @@
+    ...
+    >>> import hello
+    Hello
+    >>> with open('hello.py') as hello:
+    ...   print(hello.read())
+    ...
+    print('Hello')
+    >>> import os
+    >>> os.listdir()
+    ['hello.py']
+    >>> os.remove('hello.py')
+    >>> os.listdir()
+    []
+
+.. py:function:: open(filename, mode='r')
+
+    Returns a file object representing the file named in the argument
+    ``filename``. The mode defaults to ``'r'`` which means open for reading in
+    text mode. The other common mode is ``'w'`` for writing (overwriting the
+    content of the file if it already exists). Two other modes are available
+    to be used in conjunction with the ones describes above: ``'t'`` means
+    text mode (for reading and writing strings) and ``'b'`` means binary mode
+    (for reading and writing bytes). If these are not specified then ``'t'``
+    (text mode) is assumed. When in text mode the file object will be an
+    instance of ``TextIO``. When in binary mode the file object will be an
+    instance of ``BytesIO``.
+
+
+.. py:class:: TextIO
+
+    Instances of this class represent text files in the micro:bit's flat file
+    system.
+
+    You do not directly instantiate this class. Rather, an appropriately
+    configured instance of this class is returned by the ``open`` function
+    described above.
+
+    .. py:method:: close()
+
+        Flush and close the file. This method has no effect if the file is
+        already closed. Once the file is closed, any operation on the file
+        (e.g. reading or writing) will raise an exception.
+
+    .. py:method:: name()
+
+        Returns the name of the file the object represents. This will be the
+        same as the ``filename`` argument passed into the call to the ``open``
+        function that instantiated the object.
+
+    .. py:method:: read(size)
+
+        Read and return at most ``size`` characters from the file as a single
+        string. As a convenience, if ``size`` is unspecified or -1,
+        ``readall()`` is called. Fewer than ``size`` characters may be returned
+        if there are less than ``size`` characters remaining to be read from
+        the file.
+
+        If 0 characters are returned, and ``size`` was not 0, this indicates
+        end of file.
+
+    .. py:method:: readall()
+
+        Read all the characters from the current position in the file until
+        EOF (end of file) is reached and return them as a single string.
+
+    .. py:method:: readline(size)
+
+        Read and return one line from the file. If ``size`` is specified, at
+        most ``size`` characters will be read.
+
+        The line terminator is always ``'\n'``.
+
+    .. py:method:: writable()
+
+        Return ``True`` if the file supports writing. If ``False``, ``write()``
+        will raise ``OSError``.
+
+    .. py:method:: write(s)
+
+        Write the string ``s`` to the file and return the number of characters
+        written.
+
+
+.. py:class:: BytesIO
```

Comment:
I see.  I didn't notice that TextIo and BytesIO are using different method tables.  They should use the same tables (Ie have exactly the same methods, using the same underlying code, to save code size; that's how it is in upstream uPy).  Then the docs are the same.